app/ota_client.py

```diff
@@ -49,9 +49,70 @@ def verify_file(filename: Path, filehash: str, filesize) -> bool:
     return file_sha256(filename) == filehash
 
 
+_ExceptionWrapper = type("_ExceptionWrapper", (Exception,), dict())
```

Comment:
In my opinion, The following is natural. Dynamic generation is not required, I think.
```python
class _ExceptionWrapper(Exception):
    pass
```

Reply:
Sure!